docs/quick-actions.md

```diff
@@ -0,0 +1,81 @@
+|Shape|- Change color<br/>- Add animation<br/>- Add link<br/>- Clear filters and animation|
+|Text|- Change color<br/>- Edit text<br/>- Add animation<br/>- Add link<br/>- Clear animation|
+
+### Action functionality
+
+Reference this table when needing to know what a quick action will do when clicked.
+
+|Quick Action|Result|
+|--|--|
+|Add animation|Open and highlight the `animation` panel in the `design pane`. Focus the effect chooser dropdown.|
+|Add caption|Open and highlight the `captions and subtitles` panel in the `design pane`. Focus the input.|
+|Add link|Open and highlight the `link` panel in the `design pane`. Focus the input.|
+|Change background color|Open and highlight the `page background` panel in the `design pane`. Focus the input.|
+|Clear animation/filters and animation|Remove all filters and/or animations as specified. Display a snackbar to the user indicating that they were removed. The snackbar will contain an action button that when clicked will re-apply the styles that were removed.|
+|Edit text|Open and highlight the `text` panel in the `design pane`. Focus the first input.|
+|Insert media|Open and highlight the `media pane` in the `library`. Focus the media tab button.|
+|Insert text|Open and highlight the `text pane` in the `library`. Focus the text tab button.|
+|Replace background/media|Open and highlight one of the media panes in the `library`. The media pane opened will depend on the type of the element selected. Focus the relevant tab button.|
+
+### Visibility
+
+The quick actions menu is visible when:
+
+1. there are no elements selected or 
+2. if a single element is selected (including the background)
+
+This feature does not support selecting multiple elements, so the menu will be hidden if more than one element is selected.
+
+## Adding new quick actions
+
+Quick actions hook into the highlight api. To add a new quick action, the following steps must be completed. If you are not highlighting, skip the first two steps:
+
+1. If it doesn't already exist, add a key to `assets/src/edit-story/app/highlights/states.js` that will reference the panel/pane/element that you are trying to highlight.
+2. Attach the new highlight state to the panel/pane/element that will be focused by using the `useFocusHighlight` hook. Pass a reference to an element that should be focused if that applies.
+3. Create the quick action in the `useQuickActions` hook located at `assets/src/edit-story/app/highlights/quickActions/useQuickActions.js`
+
+## Technical notes and considerations
+
+### Focus
+
+Clicking the quick action menu should not remove focus from an element. To do this, two things were added:
+
+1. An `onMouseDown` event handler is passed to the buttons. This event handler prevents the mouse down event from unfocusing the canvas element.
+2. The context menu is given an `onMouseDown` event handler to prevent mouse  events on the menu _in between the buttons_ from bubbling up and unfocusing the element in the canvas.
```

Comment:
this was a good idea to document, i think future versions of ourselves will be thankful.